### ISSUE TITLE
SubSelectFetchTest fails

### ISSUE THREAD

**Issue by DavideD** [hibernate/hibernate-reactive#1502]
It only load the fetched collection and not all the associations that are part of the entity.

See https://github.com/hibernate/hibernate-reactive/blob/21e2c2ad7e90599a1eb8968593b6fdc7a4ea032c/hibernate-reactive-core/src/test/java/org/hibernate/reactive/SubselectFetchTest.java#L40

**Comment by blafond** [comment]
So I've tested reactive 1.1 version of `SubselectFetchTest` and as well as a `ORM_SubselectFetchTest` version and the results for `session.createQuery( "from Node n order by id", Node.class  ).getResultList() ` on both return a result set with 2 entities and neither are `Proxy`'s.

In our current reactive I ran the equivalent tests and they both return one concrete entity (Node) and one that is a `HibernateProxy` object.

Was this change expected?

**Comment by blafond** [comment]
Appears that the generated select + subselect query we used in HR 1.1 is not getting generated correctly in HR 2.x which doesn't load the collections in the primary/parent node. 

see: [SqlClientConnection.selectQuery(..)](https://github.com/hibernate/hibernate-reactive/blob/ff0f8a0792e8f5419a500ec9b40b0dd58b816b10/hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SqlClientConnection.java#L118)

OLD SQL : with paramValues array is EMPTY)
```
select 
	elements0_.node_id as node_id2_0_1_, 
	elements0_.id as id1_0_1_, 
	elements0_.id as id1_0_0_, 
	elements0_.node_id as node_id2_0_0_ 
from Element elements0_ 
        where elements0_.node_id in (select subselectf0_.id from Node subselectf0_ )
```

NEW SQL: with paramValues array has one value of  '1')
```
select 
	e1_0.node_id,
	e1_0.id
from Element e1_0 
	where e1_0.node_id=$1
```
If I  intercept the generated SQL and replace the string with the following and remove the '1' parameter, then this test succeeds. 
```
select 
	e1_0.node_id as node_id2_0_1_, 
	e1_0.id as id1_0_1_, 
	e1_0.id as id1_0_0_, 
	e1_0.node_id as node_id2_0_0_ 
from Element e1_0 
        where e1_0.node_id in (select subselectf0_.id from Node subselectf0_ )
```

So there's a step or 2 missing in the 'select' sql generation for this use-case.

**Comment by blafond** [comment]
@Davide In our 6.2.7 ORM version, the SubselectLoader isn't being set.

Looks like there was some work done on the SubselectLoader logic in [AbstractCollectionPersister.resolveSubselectLoader() ](https://github.com/hibernate/hibernate-orm/blob/0897a90a9e674d1a56633622fcd6a240d7dbda41/hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java#L727)in the main branch that may have fixed this.

**Comment by blafond** [comment]
PR merged in ORM's 6.2 branch

**Comment by DavideD** [comment]
I'm going to reopen this until we upgrade to the next ORM 6.2.
So that we don't forget to re-enable our test.

But there is no more work needed from our side to fix this issue

**Comment by DavideD** [comment]
This issue has been fixed for Hibernate Reactive 2.0, but it won't work in 2.1
I will create a separate issue
